src/sdk/namespace_fs.js

```diff
@@ -602,29 +609,36 @@ class NamespaceFS {
                 if (drain_promise) {
                     await drain_promise;
                     drain_promise = null;
+                    object_sdk.throw_if_aborted();
                 }
 
                 // write the data out to response
                 buffer_pool_cleanup = null; // cleanup is now in the socket responsibility
                 const write_ok = res.write(data, null, callback);
                 if (!write_ok) {
-                    drain_promise = stream_utils.wait_drain(res);
+                    drain_promise = stream_utils.wait_drain(res, { signal: object_sdk.abort_controller.signal })
+                        .catch(() => undefined);
```

Comment:
need to fix this to avoid swallowing errors raised from drain promise -
```suggestion
                    drain_promise = stream_utils.wait_drain(res, { signal: object_sdk.abort_controller.signal });
                    drain_promise.catch(() => undefined); // this avoids UnhandledPromiseRejection
```